ci/prow/e2e-incluster-build

```diff
@@ -30,10 +33,14 @@ echo "Waiting for the build pod to be created..."
 timeout 1m bash -c 'until oc get pods -o json | jq -er ".items[].metadata.name | select(.? | match(\"build\"))"; do sleep 1; done'
 POD_NAME=$(oc get pods -o json | jq -r '.items[].metadata.name | select(.? | match("build"))')
 
-# The build job/pod is deleted once done so we won't be able to get this info in the CI artifacts.
-echo "Print the build logs..."
-# we can't get the logs on a pod that isn't `Running` yet.
+# we can't exec a command nor get the logs on a pod that isn't `Running` yet.
 oc wait pod/${POD_NAME} --for jsonpath='{.status.phase}'=Running --timeout=60s
+
+# Check that the build secret is available to the build pod
+timeout 1m bash -c "until oc exec ${POD_NAME} -- grep super-secret-value /run/secrets/build-secret/ci-build-secret; do sleep 3; done"
```

Comment:
I think we should modify the Dockerfile to write the contents of `/run/secrets/build-secret/ci-build-secret` into the built image and grep once the ModuleLoader pod is running.

Reply:
Why not just `RUN grep super-secret-value /run/secrets/build-secret/ci-build-secret && git clone ...`?
All we really want is that the Dockerfile is able to read some secrets before running `git clone` no?!